flare/utils/parameter_helper.py

```diff
@@ -844,10 +845,11 @@ def summarize_group(self, group_type):
                         raise RuntimeError
                     self.hyps_sig[group_type] += [sig]
                     self.hyps_ls[group_type] += [ls]
-                    self.hyps_opt[group_type] += [opt_sig]
-                    self.hyps_opt[group_type] += [opt_ls]
+                    all_opt_sig += [opt_sig]
+                    all_opt_ls += [opt_ls]
                     self.logger.debug(f"   using hyper-parameters of {sig:6.2g} "
-                                      f"{ls:6.2g}")
+                                      f"{ls:6.2g} {opt_sig} {opt_ls}")
+            self.hyps_opt[group_type] = all_opt_sig + all_opt_ls
```

Comment:
So to confirm, it's okay that it's all signal hyps and _then_ all length scale params within the hyps_opt indexes? If so, this looks good to me.

Reply:
that's exactly where the bug is. The hyper-parameter array is always ordered like "sig2-1 sig2-2 ls2-1 ls2-2 sig3-1 sig3-2 ..."